ui/src/main/java/edu/wpi/grip/ui/codeGeneration/Exporter.java

```
@@ -0,0 +1,122 @@
+package edu.wpi.grip.ui.codegeneration;
+
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.Velocity;
+
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+import edu.wpi.grip.core.Pipeline;
+import edu.wpi.grip.core.Step;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.ui.codegeneration.java.TPipeline;
+
+@Singleton
+public class Exporter {
+  static{
+    Velocity.init();
+  }
+
+  public String stepNames(Pipeline pipeline) {
+    String out = "";
+    for (Step step : getSteps(pipeline)) {
+      out += step.getOperationDescription().name() + " \n";
+      out += getInputNames(step);
+      out += getOutputNames(step);
+    }
+    return out;
+  }
+
+  public static void printToFile(String string) {
+    PrintWriter writer;
+    try {
+      writer = new PrintWriter("PipelineTest.java", "UTF-8");
+      writer.println(string);
+      writer.close();
+    } catch (FileNotFoundException | UnsupportedEncodingException e) {
+      e.printStackTrace();
```

Don't do this.
You want to rethrow the exception.